sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -134,7 +135,14 @@ final class Decimal extends Ordered[Decimal] with Serializable {
   def set(bigintval: BigInteger): Decimal = {
     try {
       this.decimalVal = null
-      this.longVal = bigintval.longValueExact()
+      // TODO: Remove this once we migrate to java8 and use longValueExact() instead.
+      val isExactLong = bigintval.compareTo(BigInteger.valueOf(JLong.MAX_VALUE)) <= 0 &&
+        bigintval.compareTo(BigInteger.valueOf(JLong.MIN_VALUE)) >= 0
+      if(isExactLong) {
```

Comment:
Since we don't need the try catch, this would be much better
```scala
def set(bigintval: BigInteger): Decimal = {
    this.decimalVal = null
    // TODO: Remove this once we migrate to java8 and use longValueExact() instead.
    val isExactLong = bigintval.compareTo(LONG_MAX_BIG_INT) <= 0 &&
      bigintval.compareTo(LONG_MIN_BIG_INT) >= 0
    if (isExactLong) {
      this.longVal = bigintval.longValue()
    } else {
      throw new IllegalArgumentException(s"BigInteger ${bigintval} too large for decimal")
    }
    this._precision = DecimalType.MAX_PRECISION
    this._scale = 0
    this
  }
```